core/test/forest/CausalSurvivalForestTest.cpp

```diff
@@ -0,0 +1,50 @@
+/*-------------------------------------------------------------------------------
+  This file is part of generalized random forest (grf).
+
+  grf is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  grf is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with grf. If not, see <http://www.gnu.org/licenses/>.
+ #-------------------------------------------------------------------------------*/
+
+#include "commons/utility.h"
+#include "forest/ForestPredictor.h"
+#include "forest/ForestPredictors.h"
+#include "forest/ForestTrainer.h"
+#include "forest/ForestTrainers.h"
+#include "utilities/ForestTestUtilities.h"
+
+#include "catch.hpp"
+
+using namespace grf;
+
+TEST_CASE("causal survival forests give positive variance estimates", "[causal survival]") {
```

Comment:
When implementing survival forests, we added `SurvivalSplittingRuleTest` and `SurvivalPredictionStrategyTest`. How feasible would it be to add this sort of unit test for causal survival forests?

Reply:
I think it may be trickier with direct tests here because `SurvivalSplitting` had ground truth to compare with (log-rank test statistic for splitting, and Kaplan-Meier estimate for prediction) and this is practically the same as Instrumental/Causal forest where there are indirect test that behavior is reasonable

Reply:
That makes sense. I'm okay with the level of test coverage, especially if we have someone test it manually and 'lock in' the behavior with characterization tests. But just to brainstorm, if you're looking for more testing ideas: we could perhaps unit test some of the helper functions like `compute_eta` and the nuisance component calculations?

Reply:
Oki. Computing eta is practically just a bunch of tedious algebra: ![Screen Shot 2020-08-19 at 22 49 58](https://user-images.githubusercontent.com/7185264/90721842-7c6cb380-e26e-11ea-8316-ec039675b47f.png)

it involved taking care of a handful of edge cases which is locked in with the current characterization tests. I thought about tests for it, but did not come up with much, like plugging in oracle components for a simple DGP and seeing that the output is reasonable - it is already implicitly verified in the final simulation results in the paper.

Reply:
Also: @yifan-cui and I have done sanity checks by verifying reasonable output with DGPs over the course of this PR.

@jmzeng has also kindly beta tested the package, and as we are marking this first release "Experimental" we can incorporate improvements and suggestions as more people try it out.

Reply:
👍  it sounds like we really have confidence in correctness.